src/__tests__/pinecone.test.ts

```diff
@@ -136,12 +136,31 @@ describe('Pinecone', () => {
       });
 
       test('should throw an error if required environment variable is not set', () => {
-        delete process.env.PINECONE_API_KEY;
         expect(() => new Pinecone()).toThrow(
           "Since you called 'new Pinecone()' with no configuration object, we attempted to find client configuration in environment variables but the required environment variables were not set. Missing variables: PINECONE_API_KEY. You can find the configuration values for your project in the Pinecone developer console at https://app.pinecone.io"
         );
       });
     });
+
+    test('should log a warning if the SDK is used in a browser context', () => {
+      // Mock window simulate browser context
+      const mockWindow = {} as any;
+      global.window = mockWindow;
+
+      const warnSpy = jest
```

Comment:
Should this be an error or bug instead of a warn?

Reply:
[Discussed offline] - we want to warn rather than making the SDK unusable. If people want to use the client in a browser locally while experimenting with Pinecone that's fine, but we need to make it clear productionizing is a security risk.